linkerd/proxy/http/src/retry.rs

```diff
@@ -187,6 +187,7 @@ where
     }
 }
 
+// TODO this needs to be moved up into the application!
```

Comment:
We don't want to encode all of the possible extensions in this module. In a followup, the retry layer should take a request-cloning strategy, rather than rely on the `Request` type implementing `TryClone` itself...